app/src/main/java/net/christianbeier/droidvnc_ng/OnBootReceiver.java

```diff
@@ -54,6 +54,7 @@ public void onReceive(Context context, Intent arg1)
 
             Intent intent = new Intent(context, MainService.class);
             intent.setAction(MainService.ACTION_START);
+            intent.putExtra(MainService.EXTRA_LISTEN_INTERFACE, prefs.getString(Constants.PREFS_KEY_SETTINGS_LISTEN_INTERFACE, defaults.getListenInterface()));
```

Comment:
We cannot be sure the interface is up at that time. How to handle?

Reply:
How about something like this on the Java-side of things?
https://stackoverflow.com/questions/8919083/checking-host-reachability-availability-in-android.
It may solve the problem directly from Java, without touching the .c file at all.

Reply:
I believe that it is important to first solve the problem of intercepting network change states, after that, this should be easier.

Reply:
I've made an implementation, I haven't tested it yet though, for some reason, I'm not able to run the server at boot.
Do I need to do something in particular? I'm using a Xiaomi for testing.

Reply:
Now OnBootReceiver sends a notifications when the interface is not up, avoiding, this way, to start the service.